### PR TITLE
[terra-alert] Update docs to use mdx syntax

### PR DESCRIPTION
**Pull request by pranav300**
### Summary
Updated docs to use mdx syntax.

### CODE REVIEW

File: packages/terra-alert/package.json [hunk 1]
```diff
@@ -29,9 +29,7 @@
     "classnames": "^2.2.5",
     "prop-types": "^15.5.8",
     "terra-button": "^3.29.0",
-    "terra-doc-template": "^2.23.0",
     "terra-icon": "^3.25.0",
-    "terra-markdown": "^2.37.0",
```
Comment (emilyrohrbough) [inline packages/terra-alert/package.json]: you deleted the Markdown dep, but then its being used in the About mdx file: https://github.com/cerner/terra-core/pull/2813/files?short_path=62ce338#diff-5c324b96e5cac7b2608bf3e696e02325R73

File: packages/terra-alert/src/terra-dev-site/doc/alert/Alert.1.doc.mdx [hunk 2]
```diff
@@ -1,3 +1,23 @@
...
+import AdvisoryExample from '../example/AdvisoryExample?dev-site-example';
+import InfoExample from '../example/InfoExample?dev-site-example';
+import SuccessExample from '../example/SuccessExample?dev-site-example';
+import CustomExample from '../example/CustomExample?dev-site-example';
+import LongTextExample from '../example/LongTextExample?dev-site-example';
+import ActionExample from '../example/ActionExample?dev-site-example';
+import DismissibleExample from '../example/DismissibleExample?dev-site-example';
+import ActionAndDismissibleExample from '../example/ActionAndDismissibleExample?dev-site-example';
+
+import AlertPropsTable from '../../../../src/Alert?dev-site-props-table';
```
Comment (mjhenkes) [inline packages/terra-alert/src/terra-dev-site/doc/alert/Alert.1.doc.mdx]:
```suggestion
import AlertPropsTable from 'terra-alert/src/Alert?dev-site-props-table';
```

File: packages/terra-alert/src/terra-dev-site/doc/alert/Alert.1.doc.mdx [hunk 3]
```diff
@@ -1,3 +1,23 @@
...
+
+import ActionAndDismissableExampleDocs from '../../../../docs/action-and-dismissable-example.md';
```
Comment (mjhenkes) [inline packages/terra-alert/src/terra-dev-site/doc/alert/Alert.1.doc.mdx]: Is this import used anywhere?

File: packages/terra-alert/src/terra-dev-site/doc/alert/ChangeLog.3.doc.mdx [hunk 4]
```diff
@@ -0,0 +1,7 @@
+import { Badge } from 'terra-alert/package.json?dev-site-package';
+
+import ChangeLog from '../../../../CHANGELOG.md';
```
Comment (mjhenkes) [inline packages/terra-alert/src/terra-dev-site/doc/alert/ChangeLog.3.doc.mdx]:
```suggestion
import ChangeLog from 'terra-alert/CHANGELOG.md';
```

Comment (pranav300) [reply packages/terra-alert/src/terra-dev-site/doc/alert/ChangeLog.3.doc.mdx]: Updated here https://github.com/cerner/terra-core/pull/2813/commits/0ec3e9b2b11243ff5a0c25908ee356a9de91d5a8.

File: packages/terra-alert/src/terra-dev-site/doc/alert/Alert.1.doc.mdx (hunk 3 above)
Comment (pranav300) [reply packages/terra-alert/src/terra-dev-site/doc/alert/Alert.1.doc.mdx]: removed it here Updated here https://github.com/cerner/terra-core/pull/2813/commits/0ec3e9b2b11243ff5a0c25908ee356a9de91d5a8.

File: packages/terra-alert/src/terra-dev-site/doc/alert/Alert.1.doc.mdx (hunk 2 above)
Comment (pranav300) [reply packages/terra-alert/src/terra-dev-site/doc/alert/Alert.1.doc.mdx]: Updated here https://github.com/cerner/terra-core/pull/2813/commits/0ec3e9b2b11243ff5a0c25908ee356a9de91d5a8.

File: packages/terra-alert/package.json (hunk 1 above)
Comment (pranav300) [reply packages/terra-alert/package.json]: removed it here https://github.com/cerner/terra-core/pull/2813/commits/0ec3e9b2b11243ff5a0c25908ee356a9de91d5a8.

**Pull request merged**